core-plugin/src/main/groovy/com/github/jrubygradle/api/core/RepositoryHandlerExtension.groovy

```diff
@@ -0,0 +1,116 @@
+     */
+    RepositoryHandlerExtension(final Project project) {
+        this.project = project
+        this.ivyProxies = new IvyXmlGlobalProxyRegistry(project)
+    }
+
+    /** Create an artifact repository which will use {@link https://rubygems.org} and
+     * associate group {@code rubygems} with it.
+     *
+     * @return Artifact repository.
+     */
+    ArtifactRepository gems() {
+        bindRepositoryToProxyServer('https://rubygems.org'.toURI(), DEFAULT_GROUP_NAME)
+    }
+
+    /** Create an artifact repository which will use specified URI and
+     * associate group {@code rubygems} with it.
+     *
+     * @param uri URI of remote repository that serves up Rubygems. Any object convertible
+     * with {@code project.uri} can be provided.
+     *
+     * @return Artifact repository.
+     */
+    ArtifactRepository gems(Object uri) {
+        bindRepositoryToProxyServer(project.uri(uri), DEFAULT_GROUP_NAME)
+    }
+
+    /** Create an artifact repository which will use specified URI and
+     * associate a specified group with it.
+     *
+     * @param group Group to associate this server with.
+     * @param uri URI of remote repository that serves up Rubygems. Any object convertible
+     * with {@code project.uri} can be provided.
+     * @return Artifact repository.
+     */
+    ArtifactRepository gems(String group, Object uri) {
+        bindRepositoryToProxyServer(project.uri(uri), group)
+    }
+
+    /** Adds the legacy Torquebox Maven proxy to {@code rubygems.org}.
+     *
+     * Please note that this proxy is effectively unmaintained an no longer supported
+     * by the original creators.
+     *
+     * @return Maven artifact repository
+     */
+    MavenArtifactRepository torquebox() {
+        Action mvnConfigurator = new Action<MavenArtifactRepository>() {
+            void execute(MavenArtifactRepository mvn) {
+                mvn.url = 'http://rubygems-proxy.torquebox.org/releases'.toURI()
+            }
+        }
+        (MavenArtifactRepository)restrictToGems(
+            this.project.repositories.maven(mvnConfigurator),
+            DEFAULT_GROUP_NAME
+        )
+    }
+
+    private ArtifactRepository bindRepositoryToProxyServer(
+        URI serverUri,
+        String group
+    ) {
+        IvyXmlProxyServer proxy = ivyProxies.registerProxy(serverUri, group)
+        project.extensions.getByType(GemGroups).addGemGroup(group)
+        restrictToGems(createIvyRepo(serverUri, proxy.bindAddress), group)
+    }
+
+    @CompileDynamic
+    private IvyArtifactRepository createIvyRepo(URI server, URI bindAddress) {
+        this.project.repositories.ivy {
+            artifactPattern "${server}/downloads/[artifact]-[revision].gem"
+            ivyPattern "${bindAddress}/[organisation]/[module]/[revision]/ivy.xml"
+        }
+    }
+
+    @CompileDynamic
+    private ArtifactRepository restrictToGems(ArtifactRepository repo, String group) {
+        if (HAS_CONTENT_FEATURE) {
```

Comment:
For Gradle instances that now has the `content` DSL we restrict this repository to only to be queries if the group is `rubygems` (or another registered gem group).